src/coreComponents/codingUtilities/Section.hpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file Section.hpp
+ */
+
+#ifndef GEOS_COMMON_SECTION_HPP
+#define GEOS_COMMON_SECTION_HPP
+
+#include "common/DataTypes.hpp"
+
+namespace geos
+{
+
+class Section
+{
+public:
+
+  Section();
+
+  /**
+   * @brief Set the name of the section
+   * @param m_sectionTitle The name of the section
+   */
+  void setName( string_view m_sectionTitle );
+
+  /**
+   * @brief Add a description to the section by concatening a description name and descriptions values.
+   * @param descriptionName The description name
+   * @param args Descriptions values to be aligned.
+   * Descriptions values can be be any types and will be aligned
+   */
+  template< typename ... Args >
+  void addDescription( string const & descriptionName, Args const & ... args );
+
+  /**
+   * @brief Add a description to the section
+   * @param description The string value of the description
+   */
+  void addDescription( string const & description );
+
+  /**
+   * @brief Set the minimal width of a row
+   * @param minWidth The minimal width of the table
+   */
+  void setMinWidth( integer const & minWidth );
+
+  /**
+   * @brief Draw the first part of the section. It include the title and optionnaly, the description(s);
+   * @param os An output stream (by default, std::cout)
+   */
+  void begin( std::ostream & os = std::cout );
+
+  /**
+   * @brief Draw the last part of the section. It include the title
+   * @param oss An output stream (by default, std::cout)
+   */
+  void end( std::ostream & oss = std::cout ) const;
+
+private:
+
+  /**
+   * @brief Compute the max string size (m_rowLength) between title and the description(s)
+   * @param m_sectionTitle The table title
+   * @param descriptions The descriptions vector
+   * @return The max row length of the section
+   */
+  void computeMaxRowSize( string_view m_sectionTitle,
+                          std::vector< string > const & descriptions );
+
+  /**
+   * @brief Build a description from the name and description values
+   * @param descriptionName The decription name
+   * @param decriptionsValues The description values
+   */
+  void formatAndInsertDescriptions( std::string const & descriptionName,
```

Comment:
Can't you use `string_view` parameter here? (There are other locations in your PR)